File: src/main/java/com/example/hellowebflux/HelloWebFluxApplication.java
package com.example.hellowebflux;

import static org.springframework.web.reactive.function.server.RequestPredicates.GET;
import static org.springframework.web.reactive.function.server.RouterFunctions.route;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.ServerResponse;

@SpringBootApplication
public class HelloWebFluxApplication {

	public static void main(String[] args) {
		SpringApplication.run(HelloWebFluxApplication.class, args);
	}

	@Bean
	RouterFunction<ServerResponse> routes(HelloHandler helloHandler) {
		return route(GET("/"), helloHandler::hello);
	}
}

File: src/main/kotlin/com/example/hellowebflux/HelloWebFluxApplication.kt
package com.example.hellowebflux;

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication

@SpringBootApplication
open class HelloWebFluxApplication {

	companion object {
		@JvmStatic
		fun main(args: Array<String>) {
			SpringApplication.run(HelloWebFluxApplication::class.java, *args)
		}
	}
}
